docs/data/toolpad/core/introduction/base-concepts.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Toolpad Core - Base Concepts
+---
+
+# Base concepts
+
+<p class="description">Understand the fundamental concepts of Toolpad Core to effectively integrate and use it in your projects.</p>
+
+## Imports
+
+Toolpad Core components can be imported directly from the `@toolpad/core` package. This allows you to use them alongside your existing Material UI or other components.
+
+```tsx
+import Button from '@mui/material/Button';
+import { DashboardLayout } from '@toolpad/core/DashboardLayout';
+```
+
+## Component Hierarchy
+
+The Toolpad Core library is designed to work within your existing React application structure. The key component in this hierarchy is the `AppProvider`.
+
+### App Provider
+
+The `AppProvider` acts as a bridge between your application's runtime and Toolpad components. It should wrap your entire application or the part of your application where you want to use Toolpad components.
+
+```tsx
+import { AppProvider } from '@toolpad/core/AppProvider';
+
+function MyApp({ Component, pageProps }) {
+  return (
+    <AppProvider>
+      <Component {...pageProps} />
+    </AppProvider>
+  );
+}
+```
+
+By wrapping your application with `AppProvider`, you ensure that all other Toolpad components you use have access to the necessary context and functionality.
+
+The `AppProvider` component accepts props to configure the app's navigation, theme, branding, router, authentication, and session, like so:
+
+```tsx
+<AppProvider
+  navigation={NAVIGATION}
+  theme={theme}
+  branding={BRANDING}
+  router={router}
+  authentication={AUTHENTICATION}
+  session={session}
+>
+  {props.children}
+</AppProvider>
+```
+
+Head over to the [AppProvider](/toolpad/core/react-app-provider/) page for more details and examples of the usage of all props.
+
+:::info
+Toolpad Core doesn't handle routing itself. Instead, it's designed to integrate seamlessly with your existing routing solution, whether you're using:
+
+- Next.js App Router
+- Next.js Pages Router
+- React Router
+- Or any other routing library which implements the same interface
+
+You can pass the router implementation to the `AppProvider` component using the `router` prop. You do not need to pass this prop if you are using Next.js, since it has a file-system based router.
+
+:::
+
```

Comment:
Maybe we can add that for Next.js you don't need to set this up yourself. We have an `AppProvider` exported from `@toolpad/core/nextjs` that already sets it up correctly.

Reply:
I could add it as a separate warning from this info callout and draw specific attention to the `@toolpad/core/nextjs` export